### ISSUE TITLE
Add support for Azure Application Gateway

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#706]
Pricing: https://azure.microsoft.com/en-us/pricing/details/application-gateway/
Terraform resource: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway

**Comment by tim775** [comment]
Terraform resource azurerm_application_gateway should have

- An hourly charge based on the size and type from the sku displayed as `Gateway usage (<type>)` with unit `hours` where type is based on sku ( e.g. `basic, small`, `WAF, large`, `basic, v2`)
- A `monthly_data_processed_gb` usage parameter displayed as `Data processing (<tier>)` with unit `GB` where tier is `(0-10TB)`, `(10-40TB)`, or (`over 40TB`)
- A `monthly_v2_capacity_units` usage parameter displayed as `V2 capacity units (<type>)` with unit `CU` where type is `basic` or `WAF`

I'm not sure if a usage paramter for `monthly_outbound_data_gb` should be part of this resource or if that is captured separately (@alikhajeh1 may have a thought about this).

As always when a usage charge is possible but the key parameter is not provided, all tiers should be displayed. When the usage paremter is provided, only the tiers with a charge should be displayed.

**Comment by praveenghuge** [comment]
Hey, I can work on this

**Comment by alikhajeh1** [comment]
Thanks @tim775 and @praveenghuge!

> I'm not sure if a usage paramter for monthly_outbound_data_gb should be part of this resource or if that is captured separately (@alikhajeh1 may have a thought about this).

I would keep that separate just now since there is no specific price for it in this resource (this seems similar to what we did with AWS where we added a separate data_transfer resource)

> As always when a usage charge is possible but the key parameter is not provided, all tiers should be displayed. When the usage paremter is provided, only the tiers with a charge should be displayed.

I just reviewed what we were doing elsewhere and we only display the **first tier** (so the output is simpler for users to understand, otherwise the output becomes pretty long for new users). So maybe let's keep things consistent and only show the first tier if there is no usage key param.

One other **important** thing to note (I'll create a GH issue about this soon as we need to review a lot of code and update our contrib docs): terraform ignores case sensitivity for most things, so please ensure that:
1. case-insensitive string comparison are used. E.g. `input == "large"` is bad, `strings.ToLower(input) == "large"` is good.
2. when using any user input values as part of price filters, do a case-insenstive regex lookup. E.g.
Bad:
```
{Key: "virtualInterfaceType", ValueRegex: strPtr(fmt.Sprintf("/%s/", virtualInterfaceType))},
```

Good:
```
{Key: "virtualInterfaceType", ValueRegex: strPtr(fmt.Sprintf("/%s/i", virtualInterfaceType))},  # notice the `/i` in the regex
```

**Comment by alikhajeh1** [comment]
@praveenghuge any update on this?

**Comment by NeverovDS** [comment]
If it’s ok with Pravin, I can work on this

**Comment by alikhajeh1** [comment]
Thanks @NeverovDS!

**Comment by tim775** [comment]
- An hourly charge based on the size and type from the sku displayed as `Gateway usage (<type>, <size>)` with unit `hours` where type and size are determined from `sku.name`:  type should be one of `basic, WAF, basic v2, WAF v2`, size should only be present from non v2 instances and should be one of `small, medium, large`.   Number of hours is determined by`sku.capacity * 730`.

**Comment by alikhajeh1** [comment]
Released in v0.9.2 today.
